### DISCUSSION TITLE
Too many addons?

### DISCUSSION THREAD

**Discussion by WorldLanguages** [ScratchAddons/ScratchAddons#1663] [Discussion]
Let's discuss ways to improve user experience because of the big amount of addons we currently have (and keeps growing).
Some things I've considered (they're all related):
1. I've seen many users configure settings directly from the "addons" popup tab, which is beyond ideal. The main purpose of this popup tab is toggling addons easily, not checking through all the addons. Maybe we should make sure first time users aren't using this somehow?
2. Revisit the "recommended" tag, some addons may not deserve it, while some should have it and don't.
3. Hide low priority and beta addons from the settings page by default, inside a \<details> element? (as long as they are disabled, of course). At that point we might move easter egg addons there. We could separate addons between five \<details> elements: new addons (only expand by default on first load after update), enabled addons (sorted by category, then alphabetically, maybe), recommended (and disabled) addons, disabled addons, more addons (low priority and formerly easter eggs). If we do something like this, it's worth considering how to fit themes (which are mostly personal preference) into these categories.

**Comment by Hans5958** [comment]
## Compact design on addon popup tab (and settings page?)

One issue that I think needs attention is the padding. I get it that you need uniformity, but I think the space that one addon occupies is just too large and inefficient in this quantity. I suggest we should decrease the padding, or even better, remove the padding and just do the normal list style. Just look at other extension such as Tampermonkey.

If you still don't know what I meant, It's like the density level setting on Gmail. I prefer "Compact"  instead of the "Comfortable" ("Default" is the same as "Comfortable" in terms of the padding). See how good it is?

![image](https://user-images.githubusercontent.com/11584103/109106011-8d3c3b80-7761-11eb-8964-b5e9a65b4f2c.png)

**Reply by WorldLanguages** [reply]
We could reduce the padding.

**Comment by Hans5958** [comment]
## Only allow toggling addons on the popup tab (disallow configuring)

> I've seen many users configure settings directly from the "addons" popup tab, which is beyond ideal. The main purpose of this popup tab is toggling addons easily, not checking through all the addons. Maybe we should make sure first time users aren't using this somehow?

I prefer to just having the addons tab a place to toggle the addons, and that's it. 

I don't get it why do we allow such behavior for users, so they ignore the settings page to change some settings and to be confined to such small space. 

Take PreMiD design on the popup as an example. (There is also the settings page, if you really want)

![This is actually a .gif for troubleshooting a certain problem in PreMiD, but it does the job.](https://media.discordapp.net/attachments/607524579874832446/741949735312687124/PresenceSettings.gif)

**Reply by WorldLanguages** [reply]
By "configure the addons" I don't mean actually changing settings on them, I mean using the popup settings tab after install to check what addons we provide. For obvious reasons, for this initial revision it's way better for users to go to the full screen settings page, where they can filter by editor/website/theme and have a larger viewport.
After users are familiar with the addons they enabled, I don't think our current way of letting users change per addon settings from the popup is a bad idea, specially considering themes which autoupdate without needing a refresh.

**Reply by Hans5958** [reply]
> By "configure the addons" I don't mean actually changing settings on them, I mean using the popup settings tab after install to check what addons we provide. ....

I think you missed the point of what I said. I was saying that I don't like that users can configure the addons in the popup.

**Reply by Secret-chest** [reply]
When I first saw the idea of the addons tab, I thought of a grid layout for pinned addons, like the windows 10 action center.

**Reply by Hans5958** [reply]
@Secret-chest 

> 
> 
> When I first saw the idea of the addons tab, I thought of a grid layout for pinned addons, like the windows 10 action center.

That'll work if we talking about small, simple click toggle stuff. I don't think it applies here.

**Comment by Hans5958** [comment]
## Search filters

> Hide low priority and beta addons from the settings page by default, inside a \<details\> element? (as long as they are disabled, of course). At that point we might move easter egg addons there. We could separate addons between five \<details\> elements: new addons (only expand by default on first load after update), enabled addons (sorted by category, then alphabetically, maybe), recommended (and disabled) addons, disabled addons, more addons (low priority and formerly easter eggs). If we do something like this, it's worth considering how to fit themes (which are mostly personal preference) into these categories.

Additional idea, how about adding search filters, either something like ScratchDB, GitHub, or Discord, or with additional buttons?

----

EDIT 31/03/2021: Some examples of the search filters are listed below.

- `author:Hans5958`
- `tag:recommended` (replacing the button system)
- `since:1.12.0`
- ???

**Reply by WorldLanguages** [reply]
> either something like ScratchDB, GitHub, or Discord

What do you mean by this?

**Reply by TheColaber** [reply]
pretty sure something like versionadded: v1.10.0, madeby:TheColaber, etc.

**Reply by Hans5958** [reply]
> 
> 
> pretty sure something like versionadded: v1.10.0, madeby:TheColaber, etc.

Yeah that's what I meant

**Comment by Illodrus** [comment]
I have a couple of ideas here:

- This might be a little more work, but replacing the list with a grid and/or having an option to swap between the two could definitely be an improvement
- Moving the settings button on the popup to be a tab and having a popout symbol so it feels more clickable/discoverable to the user, especially for new users


As an afterthought this would prob cover some of the issues:
- #1211

**Reply by Hans5958** [reply]
> Moving the settings button on the popup to be a tab and having a popout symbol so it feels more clickable/discoverable to the user, especially for new users

I think that would encourage them using the addon tab even more. The settings cog icon is already universal to indicate settings.

**Reply by Hans5958** [reply]
> This might be a little more work, but replacing the list with a grid and/or having an option to swap between the two could definitely be an improvement

I think it is not appropriate for this case. It fits if there are pictures or there is a lot of content that is not wide enough. I don't know.

**Reply by Secret-chest** [reply]
Please, no videos. Only GIFs.

**Reply by Hans5958** [reply]
@Secret-chest 

> 
> 
> Please, no videos. Only GIFs.

Not sure about what you're talking about, but I would rather to have videos. A MP4 with the same content is smaller than a GIF. Want a loop? Loop it then. It's an attribute.

**Reply by Hans5958** [reply]
@RedGuy12 (please stop using the e-mail reply system, install the GitHub app already!)

> I think they mean no sound

I don't think he meant that. An MP4 could be also muted, unless that's what everyone says when referring to a silent video, regardless of file type.

**Reply by Secret-chest** [reply]
@Hans5958 @mxmou Yes I mean no sounds.

**Comment by mxmou** [comment]
What are "low priority addons"? I don't think we should hide features from users.

**Reply by mxmou** [reply]
I think the solution for them, whatever they are, is to mark non-low-priority addons as recommended and rename the "All" category to "Recommended". This means users can never see all addons at once.

I don't think beta addons need to be hidden. Maybe they could be at the end of the list of addons or in a separate category.

**Reply by WorldLanguages** [reply]
By hide, I meant putting them inside a \<details> element at the bottom of the list, that is not extended by default. They're not really hidden, just "hidden by default".

**Reply by mxmou** [reply]
@WorldLanguages Hidden by default means harder to discover. We already have a category system for addons and improving that is better than adding a new way to reduce the number of addons shown initially. I think only showing recommended addons until you switch to a more specific category, such as Themes, could be a good idea.

**Reply by WorldLanguages** [reply]
Idea is not bad but does not seem logical from a UX side? I'd expect filtering by category to show less addons, not more. Also, since you need to switch to a more specific category, non-recommended addons are still hidden by default?

**Reply by mxmou** [reply]
> Also, since you need to switch to a more specific category, non-recommended addons are still hidden by default?

I think they're much easier to discover than if you have to scroll to the bottom and toggle something to see them.

> Idea is not bad but does not seem logical from a UX side? I'd expect filtering by category to show less addons, not more.

The "All" category could be renamed to "Recommended". Or, if you still want to keep it, "Recommended" would be the default and "All" wouldn't show recommended addons before non-recommended ones.

**Reply by Secret-chest** [reply]
We should have a way to see all Addons at once.

**Comment by cobaltt7** [comment]
#1168

**Reply by WorldLanguages** [reply]
May help but does not fix the actual problem

**Comment by williamhorning** [comment]
Number 1:
I agree, we should show only enabled addons.
Number 2:
I agree
Number 3:
wdym by low priority addons?
should we sort the addons better?

**Reply by WorldLanguages** [reply]
Addons are sorted pretty nicely already, what I've suggested is grouping addons into \<details> elements: new addons, enabled addons, recommended but disabled addons, disabled addons, low priority/beta addons. This way you can easily extend and collapse each of these categories if needed, and the settings page doesn't feel that long unless you want to check every category.

**Reply by cobaltt7** [reply]
i do like an "all" tab tho, that should not be removed.

**Comment by jeffalo** [comment]
## Addon types

i shared the concept of having two addon types on the discord. quality of life, and major. (names dont really matter)

QOL addons would be the stuff that just autohides block pallete, customize block colours, 60fps, copy comment link etc.
major addons would be stuff like pause button, onion skin, folders etc (basically anything garbomuffin has touched 😜)

QOL addons would be displayed as they do right now. (after all, thats what SA was meant for), but major addons could have bigger boxes that show demo images, videos and stuff. they would show up near the top of the addon list or have their own category.

this can help with the issue of users missing out on the huge addons, and can allow for users to continue scrolling through and enabling QOL addons. (i don't know how it is for others, but for me it's a really fun experience to just scroll through and say to yourself "yes plesae yes pease yes please" while clicking all of the switches)

**Reply by Secret-chest** [reply]
No videos, only GIFs.

**Comment by cobaltt7** [comment]
Well major addons could be the recommended ones
________________________________

**Reply by Hans5958** [reply]
Reply to the post please, not make a new one

**Comment by cobaltt7** [comment]
I think they mean no sound
________________________________

**Reply by Hans5958** [reply]
> Reply to the post please, not make a new one

**Reply by mxmou** [reply]
> Reply to the post please, not make a new one

Note that this is not possible if you use email replies.

**Comment by TheColaber** [comment]
Looking back at this looks like we completed everything... The second one tho, can be implemented thru what mxmou said here: https://github.com/ScratchAddons/ScratchAddons/issues/318#issuecomment-846535127

**Comment by TheChorister** [comment]
> 1.  I've seen many users configure settings directly from the "addons" popup tab, which is beyond ideal. The main purpose of this popup tab is toggling addons easily, not checking through all the addons. Maybe we should make sure first time users aren't using this somehow?

The addon popup could have to be enabled from the fullscreen settings page, like the cloud games and scratch messaging popups.
(It could be recommended, but not enabled by default).

**Reply by WorldLanguages** [reply]
That point was already fixed, now you cannot enable addons through the addons tab, unless you use the search bar
